app/src/androidTest/java/com/github/scribeWizTeam/scribewiz/EditorTest.kt

```diff
@@ -142,6 +142,112 @@ class EditorTest {
         Assert.assertEquals(expectedNotes, actualNotes)
     }
 
+    @Test
+    fun testEditNoteInMusicXML_ReplaceNoteText() {
+        val inputXMLContent = """
+        <?xml version="1.0"?>
+        <score>
+            <note>
+                <pitch>
+                    <step>A</step>
+                </pitch>
+            </note>
+            <note>
+                <pitch>
+                    <step>B</step>
+                </pitch>
+            </note>
+            <note>
+                <pitch>
+                    <step>C</step>
+                </pitch>
+            </note>
+        </score>
+    """.trimIndent()
+        writeToFile(inputMusicXMLFile, inputXMLContent)
+
+        val expectedOutputXMLContent = """
+        <?xml version="1.0"?>
+        <score>
+            <note>
+                <pitch>
+                    <step>A</step>
+                </pitch>
+            </note>
+            <note>
+                <pitch>
+                    <step>F</step>
+                </pitch>
+            </note>
+            <note>
+                <pitch>
+                    <step>C</step>
+                </pitch>
+            </note>
+        </score>
+    """.trimIndent()
+
+        Editor.editNoteInMusicXML(inputMusicXMLFile, outputMusicXMLFile, 2, "F")
+
+        val expectedNotes = getNotes(expectedOutputXMLContent)
+        val actualNotes = getNotes(outputMusicXMLFile.readText())
+
+        Assert.assertEquals(expectedNotes, actualNotes)
+    }
+    @Test
+    fun testEditNoteInMusicXML_CopyAttributes() {
+        val inputXMLContent = """
+        <?xml version="1.0"?>
+        <score>
+            <note duration="4" type="quarter">
+                <pitch>
+                    <step>A</step>
+                </pitch>
+            </note>
+            <note duration="4" type="quarter">
+                <pitch>
+                    <step>B</step>
+                </pitch>
+            </note>
+            <note duration="4" type="quarter">
+                <pitch>
+                    <step>C</step>
+                </pitch>
+            </note>
+        </score>
+    """.trimIndent()
+        writeToFile(inputMusicXMLFile, inputXMLContent)
+
+        Editor.editNoteInMusicXML(outputMusicXMLFile, inputMusicXMLFile, 2, "F")
+
+        val outputXMLContent = outputMusicXMLFile.readText()
+        val inputNoteAttributes = extractNoteAttributes(inputXMLContent)
+        val outputNoteAttributes = extractNoteAttributes(outputXMLContent)
+
+        //println("Input note attributes: $inputNoteAttributes")
```

Comment:
Maybe you can remove debug prints here

Reply:
Yes good catch, I will commit on the next branch and remove these.